### PR TITLE
fix(tests): make testNetworkInit wait for synced chains

### PR DESCRIPTION
**Pull request by ghost**
In this PR we change `testNetworkInit` simulation test to wait for `lnd-ltc` and `lnd-btc` to be synced to the chain before proceeding. I also increased the test timeout to 30 seconds as I was occasionally reaching the 5 second limit.

### CODE REVIEW

File: test/simulation/tests.go [hunk 1]
```diff
@@ -21,23 +21,23 @@ import (
...
-		ht.assert.Equal(res.Lnd["LTC"].Chains[0], "litecoin")
-
-		// Set the node public key.
-		node.SetPubKey(res.NodePubKey)
-
-		// Add pair to the node.
-		ht.act.addPair(node, "LTC", "BTC", xudrpc.AddCurrencyRequest_LND)
+		for {
+			// retry interval
+			time.Sleep(100 * time.Millisecond)
```
Comment (moshababo) [inline test/simulation/tests.go]: Better to start with an initial attempt, and do retries on failure. 

Also maybe limit the retries/total time below the default test timeout, specially if you increased it to 30 sec.

Comment (erkarl) [reply test/simulation/tests.go]: Good point. I've changed the code to do the initial attempt first and added a timeout of 10 second. Also lowered the context timeout to 15 seconds.

**Review by sangaman — APPROVED** [review]
Awesome, thanks for figuring this out.

**Review by moshababo — APPROVED** [review]

**Comment by moshababo** [comment]
Ready for merging?

**Comment by moshababo** [comment]
Ready to merge?

**Comment by sangaman** [comment]
Yes I think so if you're ok with your 2nd comment above, making code more go idiomatic.

**Comment by erkarl** [comment]
@moshababo looks like your suggested way of doing the retry/timeout link does not point to the correct loc. Can you provide a new example?

**Comment by moshababo** [comment]
Yes it changed a bit, here's the new link. https://github.com/ExchangeUnion/xud/blob/master/test/simulation/xudtest/node.go#L215.

It's just a suggestion, we can also merge as-is if it's a hassle.

**Comment by erkarl** [comment]
Thanks for the updated link @moshababo - I'm going to merge this as is because I'm now having trouble reproducing the original issue locally which makes it very time consuming to test the changes.

**Pull request merged**